lib/store/src/csf/testing-utils/index.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  isExportStory,
+  AnyFramework,
+  AnnotatedStoryFn,
+  StoryAnnotations,
+  ComponentAnnotations,
+  ProjectAnnotations,
+  Args,
+  StoryContext,
+  Parameters,
+} from '@storybook/csf';
+
+import { composeConfigs } from '../composeConfigs';
+import { prepareStory } from '../prepareStory';
+import { normalizeStory } from '../normalizeStory';
+import { HooksContext } from '../../hooks';
+import { normalizeComponentAnnotations } from '../normalizeComponentAnnotations';
+import { getValuesFromArgTypes } from '../getValuesFromArgTypes';
+import { normalizeProjectAnnotations } from '../normalizeProjectAnnotations';
+import type { CSFExports, ComposedStoryPlayFn } from './types';
+
+export * from './types';
+
+let GLOBAL_STORYBOOK_PROJECT_ANNOTATIONS = {};
+
+export function setGlobalConfig<TFramework extends AnyFramework = AnyFramework>(
+  projectAnnotations: ProjectAnnotations<TFramework> | ProjectAnnotations<TFramework>[]
+) {
+  // deprecate this
+  setProjectAnnotations(projectAnnotations);
+}
+
+export function setProjectAnnotations<TFramework extends AnyFramework = AnyFramework>(
+  projectAnnotations: ProjectAnnotations<TFramework> | ProjectAnnotations<TFramework>[]
+) {
+  GLOBAL_STORYBOOK_PROJECT_ANNOTATIONS = Array.isArray(projectAnnotations)
+    ? composeConfigs(projectAnnotations)
+    : projectAnnotations;
+}
+
+interface ComposeStory<TFramework extends AnyFramework = AnyFramework, TArgs extends Args = Args> {
+  (
+    storyAnnotations: AnnotatedStoryFn<TFramework, TArgs> | StoryAnnotations<TFramework, TArgs>,
+    componentAnnotations: ComponentAnnotations<TFramework, TArgs>,
+    projectAnnotations: ProjectAnnotations<TFramework>,
+    exportsName?: string
+  ): {
+    (extraArgs: Partial<TArgs>): TFramework['storyResult'];
+    storyName: string;
+    args: Args;
+    play: ComposedStoryPlayFn;
+    parameters: Parameters;
+  };
+}
+
+export function composeStory<
+  TFramework extends AnyFramework = AnyFramework,
+  TArgs extends Args = Args
+>(
+  storyAnnotations: AnnotatedStoryFn<TFramework, TArgs> | StoryAnnotations<TFramework, TArgs>,
+  componentAnnotations: ComponentAnnotations<TFramework, TArgs>,
+  projectAnnotations: ProjectAnnotations<TFramework> = GLOBAL_STORYBOOK_PROJECT_ANNOTATIONS,
+  defaultConfig: ProjectAnnotations<TFramework> = {},
+  exportsName?: string
+) {
+  if (storyAnnotations === undefined) {
+    throw new Error('Expected a story but received undefined.');
+  }
+
+  // @TODO: Support auto title
+  // eslint-disable-next-line no-param-reassign
+  componentAnnotations.title = componentAnnotations.title ?? 'ComposedStory';
```

Comment:
There's a current limitation to this implementation because it doesn't calculate the title in case users are using auto title. This causes normalizeComponentAnnotations to fail:

![image](https://user-images.githubusercontent.com/1671563/160447794-f8cbbd34-9d51-4283-92f0-e0e8da5842d7.png)

So I added this workaround for now

Reply:
I guess this is fine. In cases like this (and e.g. external docs) we cannot know what the auto-title is, which has implications for story id, but I guess there's nothing we can do about it :/